### PR TITLE
Type req and add opacity as query param 

### PR DESCRIPTION
**Pull request by eraviolo**
opacity: from 0 to 1, default 1

### CODE REVIEW

File: src/server/api/geo/getForestAgreementLayer.ts [hunk 1]
```diff
@@ -6,33 +6,54 @@ import { ForestSource } from '@meta/geo'
 import { GeoController } from '@server/controller/geo'
 import Requests from '@server/utils/requests'
 
-export const getForestAgreementLayer = async (req: Request, res: Response) => {
+export const getForestAgreementLayer = async (
+  req: Request<
```
Comment (sorja) [inline src/server/api/geo/getForestAgreementLayer.ts:10]: Not sure if it's valid for your case but you can use also CycleRequest?

Comment (sorja) [reply src/server/api/geo/getForestAgreementLayer.ts:10]: Found under `meta/`

File: src/server/api/geo/getForestAgreementLayer.ts [hunk 2]
```diff
@@ -6,33 +6,54 @@ import { ForestSource } from '@meta/geo'
...
+    { countryIso: CountryIso },
+    never,
+    never,
+    { gteAgreementLevel: number; gteHansenTreeCoverPerc: number; layer: Array<string>; opacity: number }
```
Comment (sorja) [inline src/server/api/geo/getForestAgreementLayer.ts]: Is is better to defiine layer as `Array<ForestSource` ? Then you dont need to cast it later to `unknown` and `ForestSource`

File: src/server/api/geo/getForestAgreementLayer.ts [hunk 3]
```diff
@@ -6,33 +6,54 @@ import { ForestSource } from '@meta/geo'
...
+      layer.forEach((layerId: string) => {
+        if (sourceOptions.includes(layerId as unknown as ForestSource)) {
+          if (
+            layerId === ForestSource.Hansen &&
+            (gteHansenTreeCoverPerc === undefined ||
+              Number.isNaN(Number(gteHansenTreeCoverPerc)) ||
+              gteHansenTreeCoverPerc < 0 ||
+              gteHansenTreeCoverPerc > 100)
+          )
+            throw Error(`Not valid Hansen tree cover percentage 0-100: ${gteHansenTreeCoverPerc}`)
```
Comment (sorja) [inline src/server/api/geo/getForestAgreementLayer.ts:35]: If you `throw Error` I think the whole node/express server shuts down
Better to use `Requests.sendErr`

File: src/server/api/geo/getForestAgreementLayer.ts (hunk 1 above)
Comment (eraviolo) [reply src/server/api/geo/getForestAgreementLayer.ts:10]: I think for now we are not linking to a specific cycle, (only using ISO to clip country) but is something to be discussed.

File: src/server/api/geo/getForestAgreementLayer.ts (hunk 2 above)
Comment (eraviolo) [reply src/server/api/geo/getForestAgreementLayer.ts]: Yes, changed to Array<ForestSource>

File: src/server/api/geo/getForestAgreementLayer.ts (hunk 3 above)
Comment (eraviolo) [reply src/server/api/geo/getForestAgreementLayer.ts:35]: In these cases the errors are catched below and Request.sendErr is called

**Review by minotogna — APPROVED** [review]

**Review by sorja — APPROVED** [review]

**Pull request merged**